components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.listener;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+/**
+ * Log audit events in scope management.
+ */
+public class ScopeManagementAuditLogger extends AbstractEventHandler {
+
+    private static final Log audit = CarbonConstants.AUDIT_LOG;
+    private static final String AUDIT_MESSAGE = "Initiator : %s | Action : %s | Target : %s | Data : { %s } | Result" +
+            " : %s ";
+    private static final String SUCCESS = "Success";
+
+    private static final Log log = LogFactory.getLog(ScopeManagementAuditLogger.class);
+
+    /**
+     * This handles the OAuth scope related operations that are subscribed and publish audit logs for those operations.
+     *
+     * @param event Event.
+     * @throws IdentityEventException IdentityEventException.
+     */
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        int tenantId = (int) event.getEventProperties().get(IdentityEventConstants.EventProperty.TENANT_ID);
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        if (log.isDebugEnabled()) {
+            log.debug(event.getEventName() + " event received to ScopeManagementAuditLogger for the " +
+                    "tenant: " + tenantDomain);
+        }
+
+        if (IdentityEventConstants.Event.POST_ADD_SCOPE.equals(event.getEventName())) {
+            String scopeName =
+                    (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.SCOPE_NAME);
+            String initiator = getInitiatorUsername(tenantDomain);
```

Comment:
These two lines seem to be redundant for all events. How about taking them out of the if block